bee-protocol/src/worker/milestone_validator.rs

```diff
@@ -91,41 +97,50 @@ where
     type Error = WorkerError;
 
     fn dependencies() -> &'static [TypeId] {
-        Box::leak(Box::from(vec![TypeId::of::<MilestoneSolidifierWorker>()]))
+        Box::leak(Box::from(vec![
+            TypeId::of::<MilestoneSolidifierWorker>(),
+            TypeId::of::<TangleWorker>(),
+        ]))
     }
 
-    async fn start(node: &N, config: Self::Config) -> Result<Self, Self::Error> {
+    async fn start(node: &mut N, config: Self::Config) -> Result<Self, Self::Error> {
         let (tx, rx) = flume::unbounded();
         let milestone_solidifier = node.worker::<MilestoneSolidifierWorker>().unwrap().tx.clone();
 
-        let validate = match config {
-            SpongeKind::Kerl => |hash| validate_milestone::<N, Kerl, WotsPublicKey<Kerl>>(hash),
-            SpongeKind::CurlP27 => |hash| validate_milestone::<N, CurlP27, WotsPublicKey<CurlP27>>(hash),
-            SpongeKind::CurlP81 => |hash| validate_milestone::<N, CurlP81, WotsPublicKey<CurlP81>>(hash),
-        };
```

Comment:
Just curious, why was there a need to match in the task ?

Reply:
Ah, it's been made async

Reply:
`validate_milestone` became `async`